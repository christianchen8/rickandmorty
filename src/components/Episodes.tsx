"use client";

import { useShallow } from "zustand/react/shallow";
import { useCharacterStore } from "@/stores/characters";
import { SingleEpisode } from "./SingleEpisode";

export function Episodes({ position }: { position: string }) {
  const [selectedFirstCharacter, selectedSecondCharacter] = useCharacterStore(
    useShallow((state) => [
      state.selectedFirstCharacter,
      state.selectedSecondCharacter,
    ])
  );

  const character =
    position === "first" ? selectedFirstCharacter : selectedSecondCharacter;

  return (
    <div className="w-full  h-[40vh] ">
      <div className="border-b h-[6vh]">
        <h1 className="text-black font-semibold text-2xl px-4 py-2">
          Character {position === "first" ? "#1" : "#2"} - Only Episodes
        </h1>
      </div>

      <div className="h-[34vh] w-full px-4 py-2 overflow-y-scroll no-scrollbar">
        {character?.episode.map((episode) => {
          return <SingleEpisode key={episode} apiUrl={episode} />;
        })}
      </div>
    </div>
  );
}
