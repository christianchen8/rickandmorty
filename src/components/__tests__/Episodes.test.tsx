import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import { Episodes } from "../Episodes";

describe("Episodes", () => {
  // it("should render correctly", () => {
  //   const { container } = render(<Episodes position="" />);
  //   expect(container).toMatchSnapshot();
  // });

  it("should renders a heading", () => {
    render(<Episodes position="" />);

    const heading = screen.getByRole("heading", { level: 1 });
    expect(heading).toBeInTheDocument();
  });
});
