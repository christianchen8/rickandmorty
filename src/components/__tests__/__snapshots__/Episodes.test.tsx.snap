// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Episodes renders correctly 1`] = `
<div>
  <div
    class="w-full  h-[40vh] "
  >
    <div
      class="border-b h-[6vh]"
    >
      <h1
        class="text-black font-semibold text-2xl px-4 py-2"
      >
        Character 
        #2
         - Only Episodes
      </h1>
    </div>
    <div
      class="h-[34vh] w-full px-4 py-2 overflow-y-scroll no-scrollbar"
    />
  </div>
</div>
`;

exports[`Episodes should render correctly 1`] = `
<div>
  <div
    class="w-full  h-[40vh] "
  >
    <div
      class="border-b h-[6vh]"
    >
      <h1
        class="text-black font-semibold text-2xl px-4 py-2"
      >
        Character 
        #2
         - Only Episodes
      </h1>
    </div>
    <div
      class="h-[34vh] w-full px-4 py-2 overflow-y-scroll no-scrollbar"
    />
  </div>
</div>
`;
